### PR TITLE
DDF-1124 added dependency list to documentation

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?

Adds maven-generated, formatted dependency list to documentation in an appendix.

#### Who is reviewing it?

@mcalcote @brianfelix @gordocanchola @vinamartin @clockard @taz77 @cruzj6 

#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).

[Docs](https://github.com/orgs/codice/teams/docs)

#### Choose 2 committers to review/merge the PR.

@coyotesqrl
@figliold
@lessarderic
@pklinef
@shaundmorris

#### How should this be tested? (List steps with links to updated documentation)

- Verify list is formatted correctly 

#### What are the relevant tickets?

[DDF-1124](https://codice.atlassian.net/browse/DDF-1124)
([DDF-2489](https://codice.atlassian.net/browse/DDF-2489))

#### Screenshots (if appropriate)

[draft-documentation.pdf](https://github.com/codice/ddf/files/652404/draft-documentation.pdf)

#### Checklist:
- [x] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: distribution/docs/src/main/resources/_contents/_dependency-list/ddf-dependency-list.adoc [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+* org.parboiled:parboiled-java:jar:1.1.7
+* org.quartz-scheduler:quartz:jar:2.1.7
+* org.rrd4j:rrd4j:jar:2.2
+* org.slf4j:slf4j-api:jar:1.7.1
+* org.slf4j:slf4j-api:jar:1.7.12
+* org.slf4j:slf4j-ext:jar:1.7.1
+* org.slf4j:slf4j-log4j12:jar:1.7.7
+* org.slf4j:slf4j-simple:jar:1.7.1
+* org.slf4j:slf4j-simple:jar:1.7.5
+* org.spockframework:spock-core:jar:1.0-groovy-2.4
```
Comment (pklinef) [inline distribution/docs/src/main/resources/_contents/_dependency-list/ddf-dependency-list.adoc]: Should we include dependencies that are used for testing and not part of the final distribution?

File: distribution/docs/src/main/resources/_contents/_dependency-list/ddf-dependency-list.adoc [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+* org.springframework.ldap:spring-ldap-core:jar:1.3.2.RELEASE
+* org.springframework.osgi:spring-osgi-core:jar:1.1.0
+* org.springframework:spring-core:jar:4.2.5.RELEASE
+* org.taktik:mpegts-streamer:jar:0.1.0_1
+* org.twitter4j:twitter4j-core:jar:4.0.4
```
Comment (pklinef) [inline distribution/docs/src/main/resources/_contents/_dependency-list/ddf-dependency-list.adoc]: I think this is from the SDK and is not part of the final distribution.

**Comment by pklinef** [comment]
@ricklarsen How was this list of dependencies generated?

File: distribution/docs/src/main/resources/_contents/_dependency-list/ddf-dependency-list.adoc (hunk 1 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_dependency-list/ddf-dependency-list.adoc]: probably not, the test ones are supposed to be getting filtered out, but I guess it needs to be tweaked.

**Comment by ricklarsen** [comment]
@pklinef Running `mvn dependency:list` and cleaning up the results.

File: distribution/docs/src/main/resources/_contents/_dependency-list/ddf-dependency-list.adoc (hunk 1 above)
Comment (figliold) [inline distribution/docs/src/main/resources/_contents/_dependency-list/ddf-dependency-list.adoc]: Just out of curiosity, how did you get the list of dependencies?  Did you use the system directory of the distribution?

File: distribution/docs/src/main/resources/_contents/_dependency-list/ddf-dependency-list.adoc [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+* org.apache.tika:tika-parsers:jar:1.13
+* org.apache.ws.commons.axiom:axiom-api:jar:1.2.14
+* org.apache.wss4j:wss4j-bindings:jar:2.1.4
+* org.apache.wss4j:wss4j-policy:jar:2.1.4
+* org.apache.wss4j:wss4j-ws-security-common:jar:2.1.4
+* org.apache.wss4j:wss4j-ws-security-dom:jar:2.1.4
+* org.apache.wss4j:wss4j-ws-security-policy-stax:jar:2.1.4
+* org.apache.wss4j:wss4j-ws-security-stax:jar:2.1.4
+* org.asciidoctor:asciidoctorj:jar:1.5.4
+* org.assertj:assertj-core:jar:2.1.0
```
Comment (lessarderic) [inline distribution/docs/src/main/resources/_contents/_dependency-list/ddf-dependency-list.adoc]: `org.assertj:assertj-core:jar:2.1.0` is possibly a test only dependency.

File: distribution/docs/src/main/resources/_contents/_dependency-list/ddf-dependency-list.adoc [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+* org.geotools:gt-epsg-hsql:jar:8.4
+* org.geotools:gt-jts-wrapper:jar:8.4
+* org.geotools:gt-main:jar:8.4
+* org.geotools:gt-opengis:jar:8.4
+* org.geotools:gt-referencing:jar:8.4
+* org.geotools:gt-shapefile:jar:8.4
+* org.geotools:gt-xml:jar:8.4
+* org.glassfish.grizzly:grizzly-framework:jar:2.3.14
+* org.glassfish.grizzly:grizzly-http-server:jar:2.3.17
+* org.hamcrest:hamcrest-all:jar:1.3
```
Comment (lessarderic) [inline distribution/docs/src/main/resources/_contents/_dependency-list/ddf-dependency-list.adoc]: `org.hamcrest:hamcrest-all:jar:1.3`: test dependency.

File: distribution/docs/src/main/resources/_contents/_dependency-list/ddf-dependency-list.adoc (hunk 3 above)
Comment (taz77) [inline distribution/docs/src/main/resources/_contents/_dependency-list/ddf-dependency-list.adoc]: Could the test dependencies be listed separately?

**Pull request merged**